packages/teams-js/src/public/meeting.ts

```diff
@@ -925,6 +926,124 @@ export namespace meeting {
+   * This opens a meeting in a new window for the desktop app.
+   * In case of a web app, it will close the current app and open the meeting in the same tab.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @throws error if the meeting join fails, the promise will reject to an object with the error message.
+   */
+
+  export function joinMeeting(joinMeetingParams: JoinMeetingParams): Promise<void> {
+    if (!validateJoinMeetingParams(joinMeetingParams)) {
+      return Promise.reject('Invalid joinMeetingParams');
+    }
+    const { threadId, messageId, replyChainMessageId } = joinMeetingParams.chatInfo;
+
+    joinMeetingParams.joinWebUrl = new URL(joinMeetingParams.joinWebUrl);
+    const serializedJoinMeetingParams = {
+      joinWebUrl: joinMeetingParams.joinWebUrl.href,
+      subject: joinMeetingParams.subject,
+      source: joinMeetingParams.source || EventActionSource.Other,
+      chatInfo: {
+        threadId,
+        messageId: messageId || '0',
+        replyChainMessageId: replyChainMessageId || '0',
+      },
+    };
+    return new Promise<void>((resolve) => {
+      ensureInitialized(runtime);
+      resolve(
+        sendAndHandleStatusAndReasonWithVersion(
+          getApiVersionTag(ApiVersionNumber.V_2, ApiName.Meeting_JoinMeeting),
+          'meeting.joinMeeting',
+          serializedJoinMeetingParams,
+        ),
+      );
+    });
+  }
+
+  /**
+   * This function is used to check the validity of joinMeetingParams which return true if the joinMeetingParams' joinWebUrl and chatInfo are valid else false.
+   * @param joinMeetingParams The parameters for joining the meeting. {@link JoinMeetingParams}
+   * @returns false if joinMeetingParams' joinWebUrl or chatInfo is not valid else true.
+   */
+  function validateJoinMeetingParams(joinMeetingParams: JoinMeetingParams): boolean {
+    if (
+      !joinMeetingParams?.chatInfo?.threadId ||
+      !joinMeetingParams?.joinWebUrl ||
+      joinMeetingParams.chatInfo.threadId === ''
+    ) {
+      return false;
+    }
+    if (typeof joinMeetingParams.joinWebUrl === 'string' && joinMeetingParams.joinWebUrl === '') {
+      return false;
+    }
+
+    if (joinMeetingParams.joinWebUrl instanceof URL && joinMeetingParams.joinWebUrl.href === '') {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Contains information associated with parameters required for joining the Microsoft Teams meetings.
+   * More details can be found at:
+   * [Online Meeting Base - Microsoft Graph v1.0](https://learn.microsoft.com/en-us/graph/api/resources/onlinemeetingbase?view=graph-rest-1.0)
+   */
+  export interface JoinMeetingParams {
+    /** The join URL of the online meeting. */
+    joinWebUrl: URL | string;
+    /** The chat information associated with this online meeting. {@link ChatInfo}*/
+    chatInfo: ChatInfo;
+    /** The subject of the meeting. */
+    subject?: string;
```

Comment:
You didn't answer my question that I left earlier on this, so I'll ask it again here: What is the scenario where a developer would need to specify a subject of a meeting in order to join it? The meeting should already have a subject. We should never ask developers to pass up information that isn't needed to perform functionality.

Reply:
@TrevorJoelHarris we noticed that presently the meeting subject is being passed to start a meeting. However, when we tried to start a meeting without the subject too, it seemed to work.
There may be some specific scenario or corner case where this is required that we are not aware of, as this is handled by the calling and meeting team. However, presently the join meeting command demands it.

It is possible that subject is used for a pre warming scenario, or landing page scenario that we are unaware of. In case you would like me to find out more about this, I can do that.

Reply:
@RishabBindra looked into this more, and here's what this value is being used for: 

"`subject` is needed to populate the title in the pre join screen. When we join a meeting, we do not immediately have all the information related to the meeting. Hence when we land on the pre join screen, the title is needed for the new window. If the title is not passed, calling has a fallback to fetch it and populate it, however it is more latent."